Billing.API/Controllers/InvoiceController.cs

```diff
@@ -51,12 +46,51 @@ public async Task<IActionResult> GetInvoices([FromRoute] string origin, [FromRou
             return Ok(list);
         }
 
+        [HttpGet("/accounts/{origin}/{clientId:int:min(0)}/invoice/{filename}")]
+        public async Task<IActionResult> GetInvoices([FromRoute] string origin, [FromRoute] int clientId, [FromRoute] string filename)
+        {
+            _logger.LogDebug("Getting invoice for {0} client {1} filename {2}", origin, clientId, filename);
+
+            string clientPrefix;
+
+            try
+            {
+                clientPrefix = GetClientPrefix(origin);
+            }
+            catch
+            {
+                return BadRequest();
+            }
+
+            var fileParts = filename.Split('_');
+
+            var fileId = fileParts[2].ToLowerInvariant().ReplaceByEmpty(".pdf").ToInt32();
+
+            var response = await _invoiceService.GetInvoiceFile(clientPrefix, clientId, fileId);
+
+            if (response == null)
+                return NotFound();
+
+            return File(response, "application/pdf", filename);
+        }
+
         [HttpGet("/testSap")]
         public async Task<IActionResult> TestSapConnection()
         {
             var response = await _invoiceService.TestSapConnection();
 
             return Ok(response);
         }
+
+        private static string GetClientPrefix(string origin)
```

Comment:
What do you think about this and avoid exceptions for expected cases?

```suggestion
        private static bool TryGetClientPrefix(string origin, out string clientPrefix)
```